test/distributed/test_c10d_nccl.py

```diff
@@ -524,75 +552,71 @@ def reduce_scatter(outputs, input_lists, op):
         #   3
 
         # Sum
-        tensor_lists = [
-            [
-                torch.tensor([self.rank * self.num_gpus + i + j]).cuda(i)
-                for j in range(virtual_world_size)
-            ]
-            for i in range(self.num_gpus)
-        ]
+        tensor_lists = []
+        input_per_gpu = []
+
+        for i in range(self.world_size):
+            input_per_gpu.append(torch.tensor([self.rank + i + 1]))
+
+        for gpu in local_device_ids:
+            tensor_lists.append([t.cuda(device=gpu) for t in input_per_gpu])
 
         reduce_scatter(output, tensor_lists, c10d.ReduceOp.SUM)
 
-        for i in range(self.num_gpus):
+        for i in range(num_gpus):
             expected = torch.tensor(
                 [
-                    float(self.num_gpus * (self.num_gpus - 1) / 2)
-                    + (virtual_rank + i) * virtual_world_size
-                ]
-            )
+                    float((1 + self.world_size) * self.world_size / 2)
+                    + self.world_size * self.rank
+                ])
+
+
```

Comment:
I am ok with this change. Maybe we can update the comment above to reflect this change? Also, if we can differentiate the sum among different ranks, that will be great. (Just a nit, :) ).